analytics/src/main/java/com/segment/analytics/Underwriter.java

```diff
@@ -0,0 +1,41 @@
+package com.segment.analytics;
+
+import android.util.*;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class Underwriter {
+    private Mac mac;
+
+    public Underwriter(String key) {
+        if (key == null) {
```

Comment:
maybe we should add the non-nullable annotation here rather than defaulting to a value

Reply:
yeah. this is just for the sake of unit tests, since mocked client has the writekey as null. will change it to better way